faculty/clients/job.py

```diff
@@ -125,6 +150,73 @@ def make_job_summary(self, data):
         return JobSummary(**data)
 
 
+class InstanceSizeSchema(BaseSchema):
+    milli_cpus = fields.Float(data_key="milliCpus", required=True)
+    memory_mb = fields.Float(data_key="memoryMb", required=True)
+
+    @post_load
+    def make_instance_size(self, data):
+        return InstanceSize(**data)
+
+
+class JobParameterSchema(BaseSchema):
+    name = fields.String(required=True)
+    param_type = EnumField(
+        ParameterType, data_key="type", by_value=True, required=True
+    )
+    default = fields.String(required=True)
+    required = fields.Boolean(required=True)
+
+    @post_load
+    def make_job_parameter(self, data):
+        return JobParameter(**data)
+
+
+class JobCommandSchema(BaseSchema):
+    name = fields.String(required=True)
+    parameters = fields.List(
+        fields.Nested(JobParameterSchema, required=True), required=True
```

Comment:
I don't think this inner `required=True` has any effect since the parameter list can be empty.
```suggestion
        fields.Nested(JobParameterSchema), required=True
```